code/modules/mob/living/basic/trooper/trooper_ai.dm

```diff
@@ -0,0 +1,129 @@
+/datum/ai_controller/basic_controller/trooper
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/attack_until_dead
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/attack_obstacle_in_path/trooper,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/travel_to_point/and_clear_target/reinforce,
+	)
+
+/datum/ai_planning_subtree/basic_melee_attack_subtree/trooper
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/trooper
+
+/datum/ai_behavior/basic_melee_attack/trooper
+	action_cooldown = 1.2 SECONDS
```

Comment:
You shouldn't need to do this any more as long as the melee attack cooldown is set properly on the associated mob